composer/models/huggingface.py

```diff
@@ -84,10 +96,39 @@ def __init__(self,
 
         super().__init__()
         self.model = model
-        self.config = model.config
-        self.model_forward_args = inspect.getfullargspec(self.model.forward).args
+        self.config: PretrainedConfig = model.config
```

Comment:
This init function is getting quite long and it's hard to read. What do you think about doing a refactor here? Suggest a function per class property and a function for fast fails and checks

Reply:
cleaned up